### PR TITLE
bpo-38379:  don't claim objects are collected when they aren't

### PR DESCRIPTION
**Pull request by tim-one**
When a finalizer resurrects an object, nothing is actually collected in this run of gc.  Change the stats to reflect that truth.

https://bugs.python.org/issue38379

Automerge-Triggered-By: @tim-one

### CODE REVIEW

File: Lib/test/test_gc.py [hunk 1]
```diff
@@ -822,6 +822,68 @@ def test_get_objects_arguments(self):
         self.assertRaises(TypeError, gc.get_objects, "1")
         self.assertRaises(TypeError, gc.get_objects, 1.234)
 
+    def test_38379(self):
```
Comment (nascheme) [inline Lib/test/test_gc.py:825]: Maybe add a comment here explaining what the test is about.  "test_38379" will be a bit mysterious a few years from now.

**Comment by nascheme** [comment]
LGTM.  I don't think the performance impact is significant either and the stats are wrong without this fix.

**Comment by tim-one** [comment]
> I don't think the performance impact is significant either

There is no performance hit here - that would come with the second half of the bpo suggestions, if gc were taught how to keep on collecting despite that a finalizer resurrected some trash.  As is (with or without this PR so far), gc gives up after detecting a resurrection.  Which leads to the test case on the bpo issue, where trash keeps on building up forever and gc never make progress.

File: Lib/test/test_gc.py (hunk 1 above)
Comment (tim-one) [reply Lib/test/test_gc.py:825]: Absolutely - working with one finger goes slow 😉

**Comment by tim-one** [comment]
I have no idea how to backport this (well - or anything else).  I installed `cherry_picker` and ran the suggested commands, but even in `--dry-run` mode they blow up with stuff like

> subprocess.CalledProcessError: Command '['git', 'log', '-r', 'ecbf35f']' returned non-zero exit status 128.

So I'm giving up before I do real harm ☹️

**Pull request merged**